pkg/controller/importconfig/importconfig_controller.go

```diff
@@ -142,10 +119,95 @@ func (r *ReconcileImportConfig) Reconcile(ctx context.Context, request reconcile
 	return reconcile.Result{}, nil
 }
 
-func klusterletNamespace(managedCluster *clusterv1.ManagedCluster) string {
-	if klusterletNamespace, ok := managedCluster.Annotations[constants.KlusterletNamespaceAnnotation]; ok {
-		return klusterletNamespace
+func generateDefaultImportSecret(ctx context.Context, bootstrapKubeconfig, expiration []byte,
+	managedClusterName string, managedClusterAnnotations map[string]string,
+	clientHolder *helpers.ClientHolder) (*corev1.Secret, error) {
```

Comment:
it makes me feel that your should return map[string][]byte and generate secret once.

Reply:
Agree, fixed.